packages/11ty/_plugins/components/addComponentTag.js

```diff
@@ -19,9 +19,15 @@ module.exports = function(eleventyConfig, component, tagName) {
    * @see https://www.11ty.dev/docs/languages/javascript/#javascript-template-functions
    * @see https://www.11ty.dev/docs/languages/javascript/#relationship-to-filters-and-shortcodes
    */
-  eleventyConfig.addJavaScriptFunction(tagName, function(...args) {
-    return component(eleventyConfig, globalData, { page: this.page })(...args)
-  })
+  if (component.constructor.name === 'AsyncFunction') {
+    eleventyConfig.addJavaScriptFunction(tagName, async function(...args) {
+      return await component(eleventyConfig, globalData, { page: this.page })(...args)
+    })
+  } else {
+    eleventyConfig.addJavaScriptFunction(tagName, function(...args) {
+      return component(eleventyConfig, globalData, { page: this.page })(...args)
+    })
+  }
```

Comment:
```suggestion
  eleventyConfig.addJavaScriptFunction(tagName, async function(...args) {
    return (component.constructor.name === 'AsyncFunction')
      ? await component(eleventyConfig, globalData, { page: this.page })(...args)
      : component(eleventyConfig, globalData, { page: this.page })(...args)
  })
```

Reply:
This will make every component an async function